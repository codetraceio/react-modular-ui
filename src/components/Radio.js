import React from 'react';

import settings from '../settings';

import AbstractComponent from './AbstractComponent';
import Icon from './Icon';

export default class Radio extends AbstractComponent {
  getModifierObject() {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      disabled: this.props.disabled
    };
  }
  onChange(e) {
    if (typeof this.props.onChange === 'function') {
      this.props.onChange(this.props.value, e);
    }
  }
  render() {
    if (settings.isBackend()) {
      return (
        <div
          className={this.getBlockName('radio', this.getModifierObject())}
          role="radio"
          tabIndex="1"
          data-name={this.props.name}
          aria-checked={this.props.checked.toString()}
        >
          <div>
            <div
              className={this.getElementName('radio', 'icon', {
                notChecked: true
              })}
            >
              
            </div>
            <div
              className={this.getElementName('radio', 'icon', {
                checked: true
              })}
            >
              <Icon size={this.props.size} name='radio' />
            </div>
          </div>
          <div>{this.props.children}</div>
        </div>
      );
    }

    return (
      <div
        className={this.getBlockName('radio', this.getModifierObject())}
        data-name={this.props.name}
        tabIndex="1"
        onClick={(e) => this.onChange(e)}
      >
        <div className={this.getElementName('radio', 'icon')}>
          {this.props.checked ? (
            <Icon
              size={this.props.size}
              name="radio"
            />
          ) : null}
        </div>
        <div>{this.props.children}</div>
      </div>
    )
  }
}