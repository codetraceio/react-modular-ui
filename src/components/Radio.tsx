import * as React from "react";

import settings from "../services/settingService";

import {Modifiers, getBlockName, getElementName} from "../services/componentService";
import Icon from "./Icon";

export interface RadioProps {
  size?: string | number;
  view?: string;
  color?: string;
  disabled?: boolean;
  name?: string;
  value?: string | number | boolean;
  checked?: boolean;
  placeholder?: string;

  onChange?: (value: string | number | boolean, event: React.MouseEvent<HTMLDivElement>) => void;
}

export default class Radio extends React.PureComponent<RadioProps, {}> {
  getModifierObject(): Modifiers {
    return {
      size: this.props.size,
      view: this.props.view,
      color: this.props.color,
      disabled: this.props.disabled
    };
  }

  onChange = (e: React.MouseEvent<HTMLDivElement>) => {
    if (typeof this.props.onChange === "function") {
      this.props.onChange(this.props.value, e);
    }
  };

  render() {
    if (settings.isBackend()) {
      return (
        <div
          className={getBlockName("radio", this.getModifierObject())}
          role="radio"
          tabIndex={1}
          data-name={this.props.name}
          aria-checked={this.props.checked}
        >
          <div>
            <div
              className={getElementName("radio", "icon", {
                notChecked: true
              })}
            >

            </div>
            <div
              className={getElementName("radio", "icon", {
                checked: true
              })}
            >
              <Icon size={this.props.size} name="radio"/>
            </div>
          </div>
          <div>{this.props.children}</div>
        </div>
      );
    }

    return (
      <div
        className={getBlockName("radio", this.getModifierObject())}
        data-name={this.props.name}
        tabIndex={1}
        onClick={this.onChange}
      >
        <div className={getElementName("radio", "icon")}>
          {this.props.checked ? (
            <Icon
              size={this.props.size}
              name="radio"
            />
          ) : null}
        </div>
        <div>{this.props.children}</div>
      </div>
    );
  }
}