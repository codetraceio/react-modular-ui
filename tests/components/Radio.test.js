import React from 'react';
import renderer from 'react-test-renderer';
import Radio from '../../src/components/Radio';

describe('Radio', () => {
  it('should render Radio', () => {
    const tree = renderer.create(
      <Radio
        size="32"
        view="flat"
        color="primary"
        disabled
      >Something</Radio>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});