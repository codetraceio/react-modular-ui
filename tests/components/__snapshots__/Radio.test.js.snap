// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should render Radio 1`] = `
<div
  className="c-radio -size-32 -view-flat -color-primary -disabled"
  data-name={undefined}
  onClick={[Function]}
  tabIndex="1"
>
  <div
    className="c-radio--icon"
  />
  <div>
    Something
  </div>
</div>
`;
